pkg/v1/tarball/layer.go

```diff
@@ -68,16 +71,30 @@ func (l *layer) MediaType() (types.MediaType, error) {
 	return types.DockerLayer, nil
 }
 
+// LayerOption applies options to layer
+type LayerOption func(*layer) error
+
+// WithCompressionLevel sets the gzip compression. See `gzip.NewWriterLevel` for possible possible values.
+func WithCompressionLevel(level int) LayerOption {
+	return func(l *layer) error {
+		if level < gzip.HuffmanOnly || level > gzip.BestCompression {
```

Comment:
This seems premature, I'd probably just let it fail later.

Reply:
(In that the gzip package should be responsible for this validation, and not us)

Reply:
I'm okay with that. 

I'm curious what you would think about the `LayerOption` function signature in that case? Since I wouldn't be using `error` return type anymore should it be removed or should it be kept as some future proofing to prevent breaking the API?

Reply:
I'm pretty sure the interface doesn't matter, since it's impossible for anyone but us to implement it (I think), so we won't break anybody that depends on us (we can only break ourselves), so I'd remove the error. We can add it back later if we need.